src/measure_theory/measure_space.lean

```diff
@@ -997,6 +997,10 @@ end
 @[simp] lemma restrict_eq_zero : μ.restrict s = 0 ↔ μ s = 0 :=
 by rw [← measure_univ_eq_zero, restrict_apply_univ]
 
+@[simp] lemma measure.restrict_singleton {a : α} (ha : μ {a} = 0) :
+  μ.restrict {a} = 0 :=
+by simp only [measure.restrict_eq_zero, ha]
```

Comment:
Is there a reason `{a}` is not generalized to an arbitrary set?
Do we need this as a `simp` lemma if `restrict_eq_zero` is already a simp lemma?

Reply:
I agree this is mysterious, but `simp` doesn't work here without that lemma (and the simp linter is happy with my lemma).